File: booking_hotel_backend/src/main/java/com/group/hotel/room/service/FurnitureService.java
package com.group.hotel.room.service;

import com.group.hotel.room.dto.request.FurnitureCreateRequest;
import com.group.hotel.room.dto.request.FurnitureSearchRequest;
import com.group.hotel.room.dto.request.FurnitureUpdateRequest;
import com.group.hotel.room.dto.response.FurnitureResponse;

import java.util.List;

public interface FurnitureService {
    List<FurnitureResponse> getAll(FurnitureSearchRequest furnitureSearchRequest);
    FurnitureResponse create(FurnitureCreateRequest furnitureCreateRequest);
    FurnitureResponse update(Long id, FurnitureUpdateRequest furnitureUpdateRequest);
    void delete(Long id);
    List<String> getTypes();
}

File: booking_hotel_frontend/src/services/furnitureService.js
import axiosInstance from "./axiosInstance";

const furnitureService = {
    getAll: async (params) => {
        const response = await axiosInstance.get("/furnitures", { params });
        return response.data;
    },

    getTypes: async () => {
        const response = await axiosInstance.get("/furnitures/types");
        return response.data;
    },

    create: async (data) => {
        const response = await axiosInstance.post("/furnitures", data);
        return response.data;
    },

    update: async (id, data) => {
        const response = await axiosInstance.put(`/furnitures/${id}`, data);
        return response.data;
    },

    delete: async (id) => {
        const response = await axiosInstance.delete(`/furnitures/${id}`);
        return response.data;
    },
};

export default furnitureService;
